src/tools/perf/lib/libperf.c

```diff
@@ -1723,13 +1722,55 @@ ucx_perf_funcs_t ucx_perf_funcs[] = {
                           ucp_perf_test_dispatch, ucp_perf_thread_barrier}
 };
 
+ucs_status_t ucx_perf_allocators_init(ucx_perf_context_t *perf,
+                                      const ucx_perf_params_t *params)
+{
+    ucs_status_t status;
+
+    ucx_perf_test_init(perf, params);
+
+    if ((perf->send_allocator == NULL) || (perf->recv_allocator == NULL)) {
+        ucs_error("Unsupported memory types %s<->%s",
+                  ucs_memory_type_names[params->send_mem_type],
+                  ucs_memory_type_names[params->recv_mem_type]);
+        return UCS_ERR_UNSUPPORTED;
+    }
+
+    if (perf->send_allocator->mem_type != UCS_MEMORY_TYPE_HOST) {
+        ucs_diag("UCT tests also copy one-byte value from %s memory to "
+                 "%s send memory, which may impact performance results",
+                 ucs_memory_type_names[UCS_MEMORY_TYPE_HOST],
+                 ucs_memory_type_names[perf->send_allocator->mem_type]);
+    }
+
+    if (perf->recv_allocator->mem_type != UCS_MEMORY_TYPE_HOST) {
+        ucs_diag("UCT tests also copy one-byte value from %s recv memory "
+                 "to %s memory, which may impact performance results",
+                 ucs_memory_type_names[perf->recv_allocator->mem_type],
+                 ucs_memory_type_names[UCS_MEMORY_TYPE_HOST]);
+    }
+
+    status = perf->send_allocator->init(perf);
+    if (status != UCS_OK) {
+        return status;
+    }
+
+    if (perf->send_allocator != perf->recv_allocator) {
```

Comment:
Instead of
```
    if (perf->send_allocator != perf->recv_allocator) {
        status = perf->recv_allocator->init(perf);
        if (status != UCS_OK) {
            return status;
        }
    }

    return UCS_OK;
```

can have:
```
if (perf->send_allocator == perf->recv_allocator) {
    return UCS_OK;
}

return perf->recv_allocator->init(perf);
```
WDYT?

Reply:
Will change 👍